multiversx_sdk_cli/cli_wallet.py

```diff
@@ -89,38 +111,155 @@ def setup_parser(args: List[str], subparsers: Any) -> Any:
+        raise KnownError("The --outfile option is required when --format is specified.")
+
+    outfile = Path(outfile).expanduser().resolve()
+    if outfile.exists():
+        raise KnownError(f"File already exists, will not overwrite: {outfile}")
+
+    if format == WALLET_FORMAT_RAW_MNEMONIC:
+        outfile.write_text(mnemonic.get_text())
+    elif format == WALLET_FORMAT_KEYSTORE_MNEMONIC:
+        password = getpass.getpass("Enter a new password (for keystore):")
+        wallet = UserWallet.from_mnemonic(mnemonic.get_text(), password)
+        wallet.save(outfile)
+    elif format == WALLET_FORMAT_KEYSTORE_SECRET_KEY:
+        password = getpass.getpass("Enter a new password (for keystore):")
+        secret_key = mnemonic.derive_key()
+        wallet = UserWallet.from_secret_key(secret_key, password)
+        wallet.save(outfile, address_hrp)
+    elif format == WALLET_FORMAT_PEM:
+        secret_key = mnemonic.derive_key()
+        pubkey = secret_key.generate_public_key()
+        address = pubkey.to_address(address_hrp)
         pem_file = UserPEM(address.bech32(), secret_key)
-        pem_file_path = prepare_file(args.output_path, ".pem")
-        pem_file.save(pem_file_path)
-        logger.info(f"Pem wallet generated: {pem_file_path}")
-    if args.json:
-        json_file = prepare_file(args.output_path, ".json")
-        password = getpass.getpass("Enter a new password:")
+        pem_file.save(outfile)
+    else:
+        raise KnownError(f"Unknown format: {format}")
+
+    logger.info(f"Wallet ({format}) saved: {outfile}")
+
 
-        keyfile = convert_to_keyfile_object(secret_key.buffer, pubkey.buffer, password, None, "erd")
-        with open(output_path, 'w') as json_file:
-            dump(keyfile, json_file, indent=4)
+def convert_wallet(args: Any):
+    infile = Path(args.infile).expanduser().resolve() if args.infile else None
+    outfile = Path(args.outfile).expanduser().resolve() if args.outfile else None
+    in_format = args.in_format
+    out_format = args.out_format
+    address_index = args.address_index
+    address_hrp = args.address_hrp
 
-        logger.info(f"Json wallet generated: {json_file}")
+    if outfile and outfile.exists():
+        raise KnownError(f"File already exists, will not overwrite: {outfile}")
+
+    if infile:
+        input_text = infile.read_text()
+    else:
+        print(f"Press 'Ctrl-D' (Linux / MacOS) or 'Ctrl-Z' (Windows) when done.")
+        input_text = sys.stdin.read().strip()
+
+    mnemonic, secret_key = _load_wallet(input_text, in_format, address_index)
+    output_text = _create_wallet_content(out_format, mnemonic, secret_key, address_index, address_hrp)
+
+    if outfile:
+        outfile.write_text(output_text)
+    else:
+        print("Output:")
+        print()
+        print(output_text)
+
+
+def _load_wallet(input_text: str, in_format: str, address_index: int) -> Tuple[Optional[Mnemonic], Optional[UserSecretKey]]:
+    if in_format == WALLET_FORMAT_RAW_MNEMONIC:
```

Comment:
not sure about the required python version, but it seems like [>=3.10 supports switches](https://www.freecodecamp.org/news/python-switch-statement-switch-case-example/)

Reply:
Currently, the constraint for mxpy is `>=3.8`. I think we should postpone this change. But very good point, indeed!